pkg/api/types.go

```diff
@@ -341,14 +342,67 @@ type Result struct {
 	BuildInfo BuildInfo
 }
 
-// BuildInfo holds information about a particular step in the build process.
+// BuildInfo contains information about the build process.
 type BuildInfo struct {
+	// Steps contains details about each build step.
+	Steps map[StepName]StepInfo
+
 	// FailureReason is a camel case reason that is used by the machine to reply
 	// back to the OpenShift builder with information why any of the steps in the
-	// build, failed.
+	// build failed.
 	FailureReason FailureReason
 }
 
+// StepInfo contains details about a build step.
+type StepInfo struct {
+	// StartTime identifies when this step started.
+	StartTime time.Time
+
+	// StopTime identifies when this step completed
+	StopTime time.Time
+}
+
+// StepName is the identifier for a particular build step.
+type StepName string
+
+const (
+	// StepAssembleBuildScripts is the build step when the Assemble script is being
+	StepAssembleBuildScripts StepName = "AssembleBuildScripts"
+
+	// StepPullRuntimeImage is ...
+	StepPullRuntimeImage StepName = "PullRuntimeImage"
+
+	// StepDownloadAndExtractArtifacts is ...
+	StepDownloadAndExtractArtifacts StepName = "DownloadAndExtractArtifacts"
+
+	// StepLayeredBuild is ...
+	StepLayeredBuild StepName = "LayeredBuild"
+
+	// StepDownloadSource is ...
+	StepDownloadSource StepName = "DownloadSource"
+
+	// StepPullBuilderImage is the build step that is associated with pulling the
+	// builder image.
+	StepPullBuilderImage StepName = "PullBuilderImage"
+
+	// StepBuildDockerImage is ...
+	StepBuildDockerImage StepName = "BuildDockerImage"
+
+	// StepCommitContainer is the build step that is associated with commiting the
+	// container to the builder image.
+	StepCommitContainer StepName = "CommitContainer"
+
+	// StepPullPreviousImage is the build step that occurs when we try to pull a
+	// previous image for a incremental build.
+	StepPullPreviousImage StepName = "PullPreviousImage"
+
+	// StepRetrievePreviousArtifacts is ...
+	StepRetrievePreviousArtifacts StepName = "RetrievePreviousArtifacts"
+
+	// StepGatherInputContent is ...
+	StepGatherInputContent StepName = "GatherInputContent"
```

Comment:
This was a leftover for the tests to compile previously, will be removed in the next push